### ISSUE TITLE
elasticsearch + kibana not working

### ISSUE THREAD

**Issue by rcruzper** [kubernetes/kops#493]
Hi,

I created a cluster via kops in AWS and I want to add elasticsearch and kibana to the cluster.
I'm using these yaml files to do it: https://github.com/kubernetes/kubernetes/tree/v1.3.7/cluster/addons/fluentd-elasticsearch

For a while everything seem ok but a few minutes later the elasticsearch and kibana controllers and services disappeared and I have no clues what happended.

Am i doing something wrong?

**Comment by feniix** [comment]
@rcruzper you probably need to remove the tag `kubernetes.io/cluster-service: "true"` from each of the manifests

**Comment by rcruzper** [comment]
Thank you @feniix for your answer.
I've tried your solution and the elasticsearch pod is up now and the pod doesn't disappear, but when I try to run kibana pod this is the log:
```
2016-09-24T16:00:45.441411183Z ELASTICSEARCH_URL=http://elasticsearch-logging:9200
2016-09-24T16:00:47.671190729Z {"@timestamp":"2016-09-24T16:00:47.669Z","level":"error","message":"Service Unavailable","node_env":"production","error":{"message":"Service Unavailable","name":"Error","stack":"Error: Service Unavailable\n  at respond (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/src/lib/transport.js:235:15)\n  at checkRespForFailure (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/src/lib/transport.js:203:7)\n  at HttpConnector.<anonymous> (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/src/lib/connectors/http.js:156:7)\n  at IncomingMessage.bound (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/node_modules/lodash-node/modern/internals/baseBind.js:56:17)\n  at IncomingMessage.emit (events.js:117:20)\n  at _stream_readable.js:944:16\n  at process._tickCallback (node.js:442:13)\n"}}
2016-09-24T16:00:47.671806030Z {"@timestamp":"2016-09-24T16:00:47.671Z","level":"fatal","message":"Service Unavailable","node_env":"production","error":{"message":"Service Unavailable","name":"Error","stack":"Error: Service Unavailable\n  at respond (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/src/lib/transport.js:235:15)\n  at checkRespForFailure (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/src/lib/transport.js:203:7)\n  at HttpConnector.<anonymous> (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/src/lib/connectors/http.js:156:7)\n  at IncomingMessage.bound (/kibana-4.0.2-linux-x64/src/node_modules/elasticsearch/node_modules/lodash-node/modern/internals/baseBind.js:56:17)\n  at IncomingMessage.emit (events.js:117:20)\n  at _stream_readable.js:944:16\n  at process._tickCallback (node.js:442:13)\n"}}
```

Is there a reason why fluentd + elasticsearch + kibana is not installed by default with kops?

**Comment by feniix** [comment]
probably because a lot of people would not want to have to use fluend + elasticsearch + kibana? 
some people might prefer filebeats 
some other people might prefer graylog

**Comment by dmfenton** [comment]
I'm seeing that the kube-addon process is actually deleting the deployments and services for the cluster addons since the files don't exist in the /etc/kubernetes/addons directory

**Comment by rcruzper** [comment]
sorry,  I forgot to reply.
I added yml of elasticsearch and kibana to /etc/kubernetes/addons and they are working now.
But I don't know how to deply fluentd to send logs to elasticsearch.

**Comment by dmfenton** [comment]
Makes sense that that worked but definitely not in the spirit of this project to be adding files by hand.

**Comment by rcruzper** [comment]
I know but removing ```kubernetes.io/cluster-service: true``` elasticsearch returns status 503 and I think this is the problem :/

**Comment by dmfenton** [comment]
I ended up with basically the same solution. I simply copied the addons I wanted from https://github.com/kubernetes/kubernetes/tree/master/cluster/addons to each master. I'm thinking a more reasonable way to do that would be to have them added using a user-data function in AWS that runs on boot. However, it seems like @justinsb is leveraging a part of the Kubernetes API that is not exposed yet. That would definitely be a better way.

**Comment by chrislovecnm** [comment]
@rcruzper is this a kops issue or an app issue?

**Comment by rcruzper** [comment]
imho it's a kops/kubernetes issue because I want to execute elasticsearch and kibana without adding them to /etc/kubernetes/addons

**Comment by dmfenton** [comment]
@chrislovecnm I think it's both. With the way Kubernetes works today, Kops should add the selected addon files to the masters on boot. However, it would be preferable to be able to create an addon using the Kubernetes API.

**Comment by chrislovecnm** [comment]
@rcruzper please explain

**Comment by rcruzper** [comment]
@chrislovecnm I tried to add elasticsearch and kibana as services instead of addons but it was not possible, so I added them as addons to the master.

It would be great to add elasticsearch and kibana as an option when a cluster is created instead of doing it manually, @dmfenton idea is also a great feature.

**Comment by chrislovecnm** [comment]
At this point we only install kube-dns as a base package and have channel support coming in soon. If you could add some requirements to the roadmap issue we can look at possible solutions. I want to keep kops focused on cluster lifecycle management though, for instance we are not including CNI provider daemonset installs by default, but are configuring k8s to run in cni mode.

**Comment by chrislovecnm** [comment]
https://github.com/kubernetes/kops/issues/663 Is the roadmap issue

**Comment by vinayagg** [comment]
1. I copied fluentd-elasticsearch and cluster-monitoring addons folders to master node at /etc/kubernetes/addons . I waited 10 mins for k8s to automatically discover and install them but k8s did not install them (no of pods was still 12, kubectl cluster-info did not list new services)
- Any ideas why? I am using AWS, KOPS, k8s 1.4.6
- cluster-monitoring has subdirectories google/influxdb/standalone etc. Should I just pickup yaml files and copy to master node? If so, how come elasticsearch did not work as that folder only has valid yaml files?

2. Since that did not work, I used kubectl create -f . to add all services. With this approach I am facing following issue
addons/cluster-monitoring/influxdb/heapster-controller.yaml has following pre processing instructions in them, so they are not valid yaml files. 

{% set base_metrics_memory = "140Mi" -%}
{% set base_metrics_cpu = "80m" -%}

How should this be processed?
